test/Ocelot.UnitTests/Polly/PollyPoliciesDelegatingHandlerTests.cs

```diff
@@ -0,0 +1,373 @@
+            .Build();
+
+        var upstreamPath = new UpstreamPathTemplateBuilder()
+            .WithTemplate("/")
+            .WithContainsQueryString(false)
+            .WithPriority(1)
+            .WithOriginalValue("/").Build();
+
+        var route = new DownstreamRouteBuilder()
+            .WithQosOptions(options)
+            .WithUpstreamPathTemplate(upstreamPath).Build();
+
+        return route;
+    }
+
+    private static void ShouldHaveXunitHeaderWithNoContent(HttpResponseMessage actual, string headerName)
+    {
+        actual.ShouldNotBeNull();
+        actual.StatusCode.ShouldBe(HttpStatusCode.NoContent);
+        actual.Headers.GetValues("X-Xunit").ShouldContain(headerName);
+    }
+
+    private static void ShouldBeWrappedBy(FakeAsyncPolicy<HttpResponseMessage> policy, string wrapperName)
+    {
+        policy.Called.ShouldBeTrue();
+        policy.Times.ShouldBe(1);
+        policy.Method.ShouldNotBeNull();
+        policy.Target.ShouldNotBeNull();
+        policy.Method.DeclaringType?.DeclaringType.ShouldNotBeNull();
+        policy.Method.DeclaringType.DeclaringType.FullName.ShouldContain(wrapperName);
+        policy.Target.ToString().ShouldContain(wrapperName);
+    }
+
+    private async Task<HttpResponseMessage> InvokeAsync(string methodName)
+    {
+        var m = typeof(PollyPoliciesDelegatingHandler).GetMethod(methodName, BindingFlags.Instance | BindingFlags.NonPublic);
+        var task = (Task<HttpResponseMessage>)m.Invoke(_sut, new object[] { new HttpRequestMessage(), CancellationToken.None });
+        var actual = await task;
+        return actual;
+    }
+
+    internal class FakeAsyncPolicy<TResult> : AsyncPolicy<TResult>, IAsyncPolicy
+        where TResult : class
+    {
+        public object Result { get; private set; }
+        public string Name { get; private set; }
+
+        public int Times { get; protected set; }
+        public bool Called => Times > 0;
+        public MethodInfo Method { get; protected set; }
+        public object Target { get; protected set; }
+
+        public bool IsLast { get; set; }
+
+        public FakeAsyncPolicy(string name, object result)
+        {
+            Name = name;
+            Result = result;
+        }
+
+        protected override async Task<TResult> ImplementationAsync(Func<Context, CancellationToken, Task<TResult>> action, Context context, CancellationToken cancellationToken,
+            bool continueOnCapturedContext)
+        {
+            Times++;
+            Method = action.Method;
+            Target = action.Target;
+
+            if (IsLast)
+            {
+                var r = Result?.GetType() == typeof(TResult)
+                    ? (TResult)Result
+                    : Activator.CreateInstance<TResult>();
+                return r;
+            }
+
+            var result = await action(context, cancellationToken);
+            return result;
+        }
+
+        public new IAsyncPolicy WithPolicyKey(string policyKey)
```

Comment:
Note 3

Reply:
Fixed in https://github.com/ThreeMammals/Ocelot/pull/1753/commits/4e5f1fa72338f77de206fad812b76a666b3bc2cd ✔️